### ISSUE TITLE
Application is in OutOfSync state due to requirement to re-create PVC

### ISSUE THREAD

**Issue by sarunask** [argoproj/argo-cd#4086]
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a
question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

After installing VerneMQ with helm chart (from repo https://github.com/vernemq/docker-vernemq/tree/master/helm/vernemq) we observed that application is in consistent out of sync state after sync due to fact that PVC status is "OutOfSync (requires pruning)"

**To Reproduce**

1. Install app based on this Helm chart: https://github.com/vernemq/docker-vernemq/tree/master/helm/vernemq
2. Make initial sync. Wait till it becomes ok.
3. Make additional sync.
4. Application state becomes OutOfSync.

**Expected behavior**

After initial sync PVC shouldn't have to be re-created, unless it's size or type changes.

**Screenshots**

N/A

**Version**

```shell
argocd: v1.6.1+159674e
  BuildDate: 2020-06-19T00:42:09Z
  GitCommit: 159674ee844a378fb98fe297006bf7b83a6e32d2
  GitTreeState: clean
  GoVersion: go1.14.1
  Compiler: gc
  Platform: darwin/amd64
argocd-server: v1.6.2+3d1f37b
  BuildDate: 2020-07-31T23:45:07Z
  GitCommit: 3d1f37b0c53f4c75864dc7339e2831c6e6a947e0
  GitTreeState: clean
  GoVersion: go1.14.1
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: {Version:kustomize/v3.6.1 GitCommit:c97fa946d576eb6ed559f17f2ac43b3b5a8d5dbd BuildDate:2020-05-27T20:47:35Z GoOs:linux GoArch:amd64}
  Helm Version: version.BuildInfo{Version:"v3.2.0", GitCommit:"e11b7ce3b12db2941e90399e874513fbd24bcb71", GitTreeState:"clean", GoVersion:"go1.13.10"}
  Kubectl Version: v1.14.0
```

**Logs**

```
Paste any relevant application logs here.
```

**Comment by voldemortensen** [comment]
This is a duplicate of #2924, which will be fixed in argo 1.7 by this change in gitops-engine https://github.com/argoproj/gitops-engine/pull/112.

Basically, PVCs created with `volumeClaimTemplates` inside StatefulSets (which the vernemq helm chart uses) don't supply ownership information, so argo needed to work around that manually.

**Comment by sarunask** [comment]
Then we can close this issue.
